### PR TITLE
chore: migrate assertions in all unit tests in PrepareBoostTaskTest from jUnit to AssertJ.

### PR DESCRIPTION
**Pull request by bionanek**
## Summary:

https://github.com/facebook/react-native/issues/45596

Migrated all the assertions to use `assertThat()` function from AssertJ. 
Also updated the `prepareBoostTask_withMissingConfiguration_fails` test to use `assertThatThrownBy` to check if the tested task throws a given exception. 

## Changelog:
Migrate tests to assertj in these files:

- `packages/gradle-plugin/react-native-gradle-plugin/src/test/kotlin/com/facebook/react/tasks/internal/PrepareBoostTaskTest.kt`

[INTERNAL] [CHANGED] - Migrated PrepareBoostTaskTest from junit.Assert to assertj.core.api.Assertions.

For more details, see:
https://reactnative.dev/contributing/changelogs-in-pull-requests
-->

## Test Plan:

All tests pass when `./gradlew -p packages/gradle-plugin test` command is ran. 

image

### CODE REVIEW

**Comment by bionanek** [comment]
@cortinico hey there! Pinging for visibility 😇

File: packages/gradle-plugin/react-native-gradle-plugin/src/test/kotlin/com/facebook/react/tasks/internal/PrepareBoostTaskTest.kt [hunk 1]
```diff
@@ -19,11 +20,14 @@ class PrepareBoostTaskTest {
...
+  @Test
   fun prepareBoostTask_withMissingConfiguration_fails() {
     val task = createTestTask<PrepareBoostTask>()
-
-    task.taskAction()
+    assertThatThrownBy { task.taskAction() }
+        .isInstanceOf(IllegalStateException::class.java)
+        .hasMessage(
+            "Cannot query the value of task ':PrepareBoostTask' property 'boostVersion' because it has no value available."
+        )
```
Comment (cortinico) [inline packages/gradle-plugin/react-native-gradle-plugin/src/test/kotlin/com/facebook/react/tasks/internal/PrepareBoostTaskTest.kt:30]: nice :)

File: packages/gradle-plugin/react-native-gradle-plugin/src/test/kotlin/com/facebook/react/tasks/internal/PrepareBoostTaskTest.kt [hunk 2]
```diff
@@ -62,7 +66,7 @@ class PrepareBoostTaskTest {
     }
     task.taskAction()
 
-    assertTrue(File(output, "asm/asm.S").exists())
+    assertThat(File(output, "asm/asm.S").exists()).isTrue()
```
Comment (cortinico) [inline packages/gradle-plugin/react-native-gradle-plugin/src/test/kotlin/com/facebook/react/tasks/internal/PrepareBoostTaskTest.kt:69]:
```suggestion
    assertThat(File(output, "asm/asm.S")).exists()
```

File: packages/gradle-plugin/react-native-gradle-plugin/src/test/kotlin/com/facebook/react/tasks/internal/PrepareBoostTaskTest.kt [hunk 3]
```diff
@@ -81,7 +85,7 @@ class PrepareBoostTaskTest {
     }
     task.taskAction()
 
-    assertTrue(File(output, "boost_1.0.0/boost/config.hpp").exists())
+    assertThat(File(output, "boost_1.0.0/boost/config.hpp").exists()).isTrue()
```
Comment (cortinico) [inline packages/gradle-plugin/react-native-gradle-plugin/src/test/kotlin/com/facebook/react/tasks/internal/PrepareBoostTaskTest.kt:88]: same as above

File: packages/gradle-plugin/react-native-gradle-plugin/src/test/kotlin/com/facebook/react/tasks/internal/PrepareBoostTaskTest.kt [hunk 4]
```diff
@@ -100,6 +104,6 @@ class PrepareBoostTaskTest {
...
```
Comment (cortinico) [inline packages/gradle-plugin/react-native-gradle-plugin/src/test/kotlin/com/facebook/react/tasks/internal/PrepareBoostTaskTest.kt:107]: same as above

**Review by cortinico — CHANGES_REQUESTED** [review]

File: packages/gradle-plugin/react-native-gradle-plugin/src/test/kotlin/com/facebook/react/tasks/internal/PrepareBoostTaskTest.kt (hunk 3 above)
Comment (bionanek) [reply packages/gradle-plugin/react-native-gradle-plugin/src/test/kotlin/com/facebook/react/tasks/internal/PrepareBoostTaskTest.kt]: updated

File: packages/gradle-plugin/react-native-gradle-plugin/src/test/kotlin/com/facebook/react/tasks/internal/PrepareBoostTaskTest.kt (hunk 4 above)
Comment (bionanek) [reply packages/gradle-plugin/react-native-gradle-plugin/src/test/kotlin/com/facebook/react/tasks/internal/PrepareBoostTaskTest.kt]: updated

**Review by cortinico — APPROVED** [review]

**Pull request closed without merging**